data/src/config.rs

```diff
@@ -48,40 +30,42 @@ struct DataSource {
 }
 
 
+/// A mapping between a server resource and a local location.
+#[derive(Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct DownloadMap {
+    /// A URL string indicating the location of the resource
+    pub url: String,
+
+    /// The path to the location where the file should be saved
+    pub dest: PathBuf,
+}
+
+
 /// Downloads all files from the config file.
 ///
 /// # Panics
 ///
-/// Panics if the config file is badly formatted, when something goes wrong during download, or a
-/// write error occurs.
-pub fn download(active_sources: &Vec<&str>) {
-    let config: Config =
-        toml::from_str(include_str!("config.toml")).expect("Failed to parse config file");
+/// Panics if the config file is badly formatted.
+pub fn get_download_maps(sources: &Vec<&str>) -> Vec<DownloadMap> {
+    let mut maps: Vec<DownloadMap> = vec![];
 
-    let mut downloads = vec![];
+    let config: Config = toml::from_str(CONFIG_TOML).expect("Failed to parse sources.toml file");
 
     for (source, info) in config {
-        if !active_sources.contains(&source.as_str()) {
+        if !sources.is_empty() && !sources.contains(&source.as_str()) {
```

Comment:
I had to go thorough a good deal of thinking to check if this was correct. I'm still not fully convinced. With only one condition, the continue is fine, but I think now it'd be better to reword this into the positive and wrap the following loops rather than have the continue.